sqp/base.py

```diff
@@ -1,172 +1,379 @@
 
 import numpy as np
-from typing import Tuple, Sequence
+from typing import Tuple, Optional
 
 from quadratic.base import minimize_quadratic_problem
 from shared.minimization_problem import MinimizationProblem
 from shared.constraints import Constraint, LinearCallable, LinearConstraint, EquationType, combine_linear
 from quadratic.quadratic_problem import QuadraticProblem
-from sqp.quasi_newton_approx import damped_bfgs_updating, sr1
+from sqp.quasi_newton_approx import sr1
 
 
-# should be done
 def minimize_nonlinear_problem(
         original_problem: MinimizationProblem,
         eta=0.8, tau=0.8, tolerance=1e-3, max_iter=100_000) -> Tuple[np.ndarray, int]:
+    """
+    Performs the Line Search SQP Algorithm (18.3) to solve a non-linear constrained problem.
 
-    # we expect inequalities to be greater-than inequalities
-    prepared_constraints = [c.as_ge_if_le() for c in original_problem.constraints]
-    problem = MinimizationProblem(f=original_problem.f, n=original_problem.n, x0=original_problem.x0,
-                               solution=original_problem.solution, constraints=prepared_constraints)
+    Args:
+        original_problem: Problem that we want to minimize
+        eta: As in 18.3 - used to dampen the influence of the directional gradient in the line search
+        tau: As in 18.3 - multiplied with alpha in the line search when step-size condition is not yet fulfilled
+        tolerance: Tolerance of how close our solution needs to be in terms of the KKT-conditions
```

Comment:
I bin mir da ned ganz sicher, was des heißt, also ob des jetzt was mit ana Norm zu tun hat oder so; es is minor aber i find die Doku recht schön eigentlich und das würds bissl runden 🤗

Reply:
ja in dem fall wird die toleranz in mehreren bereichen eingesetzt und alles davon zu dokumentiern wär recht aufwendig, weils ja fürs ganze KKT-system, manchma für normen, manchma elementwise (bzw auf function value). kanns aber noch etwas ausführlicher schreibn

Reply:
Es is nur so a minor Ding, i find es passt auch so schon ganz gut eigentlich :) Nur weil i's von  der Doku ned glei verstanden hab.